### PR TITLE
BUG: Fix itkFrequencyImageRegionIterator

### PR DESCRIPTION
**Pull request by phcerdan**
Test it via recently added gtest (see #4827)

Closes #4828

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)

### CODE REVIEW

File: Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h [hunk 1]
```diff
@@ -174,8 +174,10 @@ class ITK_TEMPLATE_EXPORT FrequencyImageRegionConstIteratorWithIndex : public Im
   void
   Init()
   {
-    this->m_FrequencyOrigin = this->m_Image->GetOrigin();
-    this->m_FrequencySpacing = this->m_Image->GetSpacing();
+    for (unsigned int dim = 0; dim < ImageType::ImageDimension; ++dim) {
```
Comment (dzenanz) [inline Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h:177]: Was this set of changes necessary?

**Review by dzenanz — APPROVED** [review]
LGTM

File: Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h (hunk 1 above)
Comment (phcerdan) [reply Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h:177]: Yeah, different types

```cpp
Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h:177:29:
 error: no match for ‘operator=’ (operand types are ‘itk::FrequencyImageRegionConstIteratorWithIndex<itk::Image<double, 3> >::FrequencyType’ {aka ‘itk::Vector<double, 3>’} and
  ‘const itk::ImageBase<3>::PointType’ {aka ‘const itk::Point<double, 3>’})
  177 |     this->m_FrequencyOrigin = this->m_Image->GetOrigin();
  ```

**Comment by N-Dekker** [comment]
Thanks @phcerdan but are those two iterators really useful? Given the fact that these two did not compile in the six years of their existence, why not remove them by now?

If they are really useful (next to all the other iterator types that we have already in ITK), can you please add a test or example to show their usefulness?

File: Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h (hunk 1 above)
Comment (N-Dekker) [reply Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h:177]: Is there a specific reason why `m_FrequencyOrigin` and `m_Image->GetOrigin()` have different types? I mean, why not declare  `m_FrequencyOrigin` as an `itk::Point` as well?

Comment (phcerdan) [reply Modules/Filtering/ImageFrequency/include/itkFrequencyImageRegionConstIteratorWithIndex.h:177]: I don't exactly remember the reasons, but I think Vector has more math operations operators than Point.

All the other frequency iterators are also itk::Vector. If you feel like changing it, open an issue and let's discuss it, but it is out of the scope of this fix.

**Comment by phcerdan** [comment]
Done

**Pull request merged**